src/components/editor/index.js

```diff
@@ -0,0 +1,261 @@
+import { PropTypes } from 'react';
+import reactStamp from 'react-stamp';
+import { fromJS, List, Repeat } from 'immutable';
+import Editor from 'draft-js-plugins-editor';
+import {
+  EditorState,
+  ContentState,
+  ContentBlock,
+  convertFromHTML,
+  convertToRaw,
+  convertFromRaw,
+  RichUtils,
+  Modifier,
+  BlockMapBuilder,
+  genKey,
+  CharacterMetadata,
+} from 'draft-js';
+
+function insertFragment(editorState, fragment, entityMap) {
+  var newContent = Modifier.replaceWithFragment(
+    editorState.getCurrentContent(),
+    editorState.getSelection(),
+    fragment
+  );
+
+  return EditorState.push(
+    editorState,
+    newContent,
+    'insert-fragment'
+  );
+}
+
+const EditorComponent = (React, ...behaviours) => reactStamp(React).compose({
+  propTypes: {
+    content: PropTypes.shape({
+      entityMap: PropTypes.object.isRequired,
+      blocks: PropTypes.array.isRequired,
+    }).isRequired,
+
+    onEditStart: PropTypes.func,
+    onEditEnd: PropTypes.func,
+    onChange: PropTypes.func,
+
+    readOnly: PropTypes.bool,
+    plugins: PropTypes.array,
+  },
+
+  init() {
+    const content = convertFromRaw(this.props.content);
+    this.state = {
+      editor: EditorState.createWithContent(content),
+      readOnly: true,
+    };
+
+    this.onClick = this.onClick.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+
+    this.handleKeyCommand = this.handleKeyCommand.bind(this);
+    this.handleReturn = this.handleReturn.bind(this);
+    this.handlePastedText = this.handlePastedText.bind(this);
+
+    this.save = this.save.bind(this);
+  },
+
+  componentWillUnmount() {
+    this.deleteTimeout();
+    this.save();
+  },
+
+  onClick() {
+    this.setState({ readOnly: false }, () => {
+      this.refs.upstream && this.refs.upstream.focus();
```

Comment:
As per another comment, we assume Draft's focus function works as expected for the purpose of a *unit test*. If we want to check our `onClick` does the correct behaviour, we really only want to check that it sets state and calls the function. This is arguably a case where a unit test is the wrong solution. One could make the case that this is more appropriate for an integration test.

However, one way to make this testable is to have the second argument for `setState` be `::this._focus`, which contains the same logic. Then in our unit test, we set the component instance method `instance._focus` to a spy and ensure the spy gets called.